### PR TITLE
Copy apmelasticsearch round_tripper to work around goroutine leak

### PR DESCRIPTION
**Pull request by pebrc**
Copy the apmelasticsearch roundtripper to work around a resource leak due to the missing propagation of `CloseIdleConnections`

### CODE REVIEW

File: pkg/controller/common/tracing/round_tripper.go [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+	"path"
+	"strings"
+	"sync/atomic"
+	"unsafe"
+
+	"go.elastic.co/apm"
+	"go.elastic.co/apm/module/apmhttp"
+)
+
+// This purely exists as workaround until the issue around closing connections in apm-go-agent is fixed.
```
Comment (sebgl) [inline pkg/controller/common/tracing/round_tripper.go:23]: Can you highlight what is the original code and what is the fix?

Comment (pebrc) [reply pkg/controller/common/tracing/round_tripper.go:23]: Yes absolutely.

**Comment by pebrc** [comment]
The APM fix got already merged so we have alternatively the option to upgrade the APM libraries in #2921

**Comment by pebrc** [comment]
Closing in favor of #2921

**Pull request closed without merging**